Add doc changes to include the new fields in stack config policy (#7381)
elastic/cloud-on-k8s: docs/orchestrating-elastic-stack-applications/security/auth-configs-using-stack-config-policy.asciidoc

:page_id: auth-config-using-stack-config-policy
ifdef::env-github[]
****
link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-{page_id}.html[View this document on the Elastic website]
****
endif::[]
[id="{p}-{page_id}"]

> = Authentication policies using Elastic Stack configuration policy

Reviewer: I'm wondering if the idea of "authentication policy" makes sense here. 

```suggestion
= Managing authentication realms and role mapping using Elastic Stack configuration policy
```

Reviewer 2: Yes I think we should avoid the concept of authentication policies. It will only confuse users. I think when we introduced it internally we thought we would have a separate CRD for it but that is not the path we went down.

Reviewer 2: ```suggestion
= Managing authentication for multiple stacks using Elastic Stack configuration policy
```
just an idea

NOTE: This requires a valid Enterprise license or Enterprise trial license. Check <<{p}-licensing,the license documentation>> for more details about managing licenses.

> Starting from ECK `2.11.0` new options were added that enable adding configurations that go into `elasticsearch.yml` and `kibana.yml` to the Elastic Stack configuration policy.

Reviewer 2: I would probably phrase it a little bit more with focus on what users can do with it. 

```suggestion
ECK `2.11.0` extends the functionality of [link...]Elastic Stack configuration policies so that it becomes possible to configure Elasticsearch security realms for more than one Elastic stack at once. 
```

Author: Made the suggested change.

> There is also an option to add additional secrets that will be mounted to the Elasticsearch Pods managed by the Elastic Stack configuration policy. This can be used to configure authorization policies in Elastic Stack configuration policy. 

Reviewer: It's not clear if this documentation is about "authentication" or "authorization", which are 2 different things. (https://auth0.com/docs/get-started/identity-fundamentals/authentication-and-authorization)

Author: my bad, this should have been authentication.

The authorization policy will apply to all Elasticsearch clusters and Kibana instances managed by the Elastic Stack configuration policy.

Examples for configuring some of the authentication policies can be found below:

** <<{p}-ldap-using-stack-config-policy,LDAP authentication using Elastic Stack configuration policy>>
** <<{p}-oidc-stack-config-policy,OpenID Connect authentication using Elastic Stack configuration policy>>

> ** <<{p}-jwt-stack-config-policy,JWT authentication using Elastic Stack configuration policy>>

Reviewer 2: I wonder if we can do something so that the index structure is more hierarchical
<img width="341" alt="Screenshot 2023-12-18 at 16 08 40" src="https://github.com/elastic/cloud-on-k8s/assets/697790/00da6459-8b99-45f7-9290-fc0487337f2b">

I think what we would like is to have is a structure like 
* Custom HTTP certificate
* Rotate auto-generated credentials
* Managing authentication for multiple stacks using Elastic Stack configuration policy

and then on the right hand side the navigation for the subsections (not sure if that's possible)

Author: Let me look into this, this makes sense to me.

Author: Made the change
<img width="1553" alt="Screenshot 2023-12-18 at 11 04 44 AM" src="https://github.com/elastic/cloud-on-k8s/assets/135071014/9b285564-44f2-4dd4-98ac-2384cf8ae9d4">
